src/common/util.cpp

```diff
@@ -967,4 +967,23 @@ std::string get_nix_version_display_string()
   }
 #endif
 
+  void closefrom(int fd)
+  {
+#if defined __FreeBSD__ || defined __OpenBSD__ || defined __NetBSD__
```

Comment:
```suggestion
#if defined __FreeBSD__ || defined __OpenBSD__ || defined __NetBSD__ || defined __DragonFly__
```

DragonFly BSD has been dumb about detection in the past, this may certainly be needed.

Reply:
Needed is unlikely since the default is the manual way, which should work with any UNIX. If anyone with Dragonfly can check the headers to see if closefrom is around, I'll add it.

Reply:
>Needed is unlikely since the default is the manual way, which should work with any UNIX

I brought this up because of my experience porting kovri to dragonfly, and the definition was needed in several areas.

> If anyone with Dragonfly can check the headers to see if closefrom is around, I'll add it.

Yep, there, and their `unistd.h` is also pulled from freebsd.